logzio-sender/src/main/java/io/logz/sender/LogzioSender.java

```diff
@@ -260,16 +264,30 @@ private int sizeInBytes(List<FormattedLogMessage> logMessages) {
     }
 
     private byte[] toNewLineSeparatedByteArray(List<FormattedLogMessage> messages) {
-
         try {
             ByteArrayOutputStream byteOutputStream = new ByteArrayOutputStream(sizeInBytes(messages));
             for (FormattedLogMessage currMessage : messages) byteOutputStream.write(currMessage.getMessage());
+            if (this.doGzip){
+                return compressToGzip(byteOutputStream.toByteArray());
```

Comment:
How about decorating the byteOutputStream with GzipOutputStream, based on condition?